vtex-apps/klaviyo-pixel
# [KLAVIYO-10] Fix script for starter checkout event

Started checkout event is a script where first you have to send the user info, but the i had to:
1. Change to `$(window).on(orderFormUpdated.vtex)` in order to wait for the orderForm to be updated and then send the info to klaviyo trough _learnq.push
2. After this push i had to wait 3 second in order to ensure we don't hit any rate limits or cause any rate conditions.

below is a conversation screenshot with Klaviyo support.
![Screen Shot 2022-05-11 at 14 04 22](https://user-images.githubusercontent.com/49737670/167929113-c954f4b3-7bec-4fa6-acfc-afe19e304e8c.png)

##How to test it?
Go to https://horzeusa.myvtex.com/checkout#/shipping.
Select a product and go to checkout.
Open the browser console on the network section.
After you fill your name, last name and phone, and click on Proceed to shipping.
You should see in the browser network three calls related with Klaviyo:
1. https://a.klaviyo.com/api/onsite/identify?c=XMiUnd
2. https://a.klaviyo.com/api/track with the body request of `data: {"event":"__activity__"...`
3. https://a.klaviyo.com/api/track with the body request of `data: {"event":"Started Checkout"...`
